File: packages/targets/sources/ios/bindings/Haiku.swift
// TODO: this should also accept options.
extension Haiku {
    /**
     The `URL` of the resource, or `nil` if it could not be parsed.

     - See: [File.url](x-source-tag://File.url)
     */
    @objc public var url: URL? {
        return file.url
    }

    var file: File {
      return File(src: "haiku/\(component).html", type: "raw")
    }
}

File: examples/playground/web/diez/sources/haiku/Haiku.js
import {File} from '../file/File';

export class Haiku {
  constructor ({component} = {component: ''}) {
    this.component = component;
  }

  update (payload) {
    if (!payload) {
      return;
    }

    if (payload.component !== undefined) {
      this.component = payload.component;
    }
  }
}

Object.defineProperty(Haiku.prototype, 'file', {
  get () {
    return new File({src: `haiku/${this.component}.html`});
  },
});

Object.defineProperty(Haiku.prototype, 'url', {
  get () {
    return this.file.url;
  },
});
